test/Microsoft.AspNet.Mvc.FunctionalTests/Logging/LoggingExtensions.cs

```diff
@@ -0,0 +1,248 @@
+        }
+
+        /// <summary>
+        /// Gets a flattened list of logs that are logged for a given request 
+        /// </summary>
+        /// <param name="activities"></param>
+        /// <param name="requestTraceId">The "RequestTraceId" query parameter value</param>
+        /// <returns>List of log messages</returns>
+        public static IEnumerable<LogInfoDto> GetLogs(this IEnumerable<ActivityContextDto> activities,
+                                                     string requestTraceId)
+        {
+            return activities.FilterByRequestTraceId(requestTraceId)
+                             .GetLogs();
+        }
+
+        /// <summary>
+        /// Get a flattened list of logs present within a given scope and that are logged 
+        /// for a given request
+        /// </summary>
+        /// <param name="activities"></param>
+        /// <param name="requestTraceId">The "RequestTraceId" query parameter value</param>
+        /// <param name="scopeName"></param>
+        /// <returns></returns>
+        public static IEnumerable<LogInfoDto> GetLogs(this IEnumerable<ActivityContextDto> activities,
+                                                     string requestTraceId,
+                                                     string scopeName)
+        {
+            return activities.FilterByRequestTraceId(requestTraceId)
+                             .GetLogsUnderScope(scopeName);
+        }
+
+        /// <summary>
+        /// Filters log activities based on the given request.
+        /// </summary>
+        /// <param name="activities"></param>
+        /// <param name="requestTraceId">The "RequestTraceId" query parameter value</param>
+        /// <returns></returns>
+        public static IEnumerable<ActivityContextDto> FilterByRequestTraceId(this IEnumerable<ActivityContextDto> activities,
+                                                                                string requestTraceId)
+        {
+            return activities.Where(activity => activity.RequestInfo != null
+                        && string.Equals(GetQueryValue(activity.RequestInfo.Query, RequestTraceIdQueryKey),
+                                        requestTraceId,
+                                        StringComparison.OrdinalIgnoreCase));
+        }
+
+        /// <summary>
+        /// Compares two trees and verifies if the scope nodes are equal
+        /// </summary>
+        /// <param name="root1"></param>
+        /// <param name="root2"></param>
+        /// <returns></returns>
+        public static bool AreScopesEqual(ScopeNodeDto root1, ScopeNodeDto root2)
+        {
+            if (root1 == null && root2 == null) return true;
+
+            if (root1 == null || root2 == null) return false;
+
+            if(!string.Equals(root1.State?.ToString(), root2.State?.ToString())
+                || root1.Children.Count != root2.Children.Count)
+            {
+                return false;
```

Comment:
Think about changing this to be `LoggingAssert.ScopesEqual` and then replacing a bunch of this stuff with `Assert.Foo(...)`

A method like this will always be consumed from tests like `Assert.True(LoggingExtensions.AreScopesEqual(....))`. It's verbose and we should try to avoid `Assert.True` as much as possible, as the error messages you get back are useless.